gpMgmt/bin/gpexpand

```diff
@@ -773,8 +773,19 @@ class SegmentTemplate:
 
     def _distribute_tarfile(self):
         """Distributes template tar file to hosts"""
+        localHostname = getLocalHostname()
+        # NOTE: Currently, GP's scp transfers relative file paths into the home
+        # directory by default.
+        pathsAreSame = os.path.realpath(self.segTarFile) == os.path.join(getHomePath(), self.schema_tar_file)
+
         for host in self.hosts:
-            logger.debug('Copying tar file to %s' % host)
+            # If the source and the destination are the same on the local
+            # machine, scp may corrupt the file. Don't overwrite it.
+            if host in ('localhost', localHostname) and pathsAreSame:
```

Comment:
I think that this determination of the right host is not very reliable. What if someone defines several hosts in `/etc/hosts` that point to `localhost`? And these hosts are listed in `self.hosts`.

Reply:
Okay. What about this situation? One host in `/etc/hosts` points to `127.0.0.1`, and another one points to `10.0.0.1`, where `10.0.0.1` is the address of the current machine. And both of these hosts are in `self.hosts`.

Reply:
`self.hosts` is filled by running `/bin/hostname` with ssh on each host. Do you think this situation is likely and the same host can somehow be resolved to different IPs? How would you like the check to look like?

Reply:
> How would you like the check to look like?

Maybe we need to check the file existence. Or we can also calculate `md5`.

Reply:
The normal operation of `gpexpand` assumes that if we reached this part, the tar file exists, was created by this very `gpexpand` usage and is recent.

Reply:
> the tar file exists

I meant the existence of a file (and not just a tar archive) on a remote host.

Reply:
What's wrong with overwriting the file on the remote host if it's needed? The problem only affects the local host.

Reply:
> overwriting the file on the remote host if it's needed

In order to avoid rewriting again, we can compare, for example, md5.

Reply:
That seems like unnecessary work and is not related to this task as well. Maybe bring it up as a separate one?